src/node/node_state.h

```diff
@@ -1405,41 +1414,45 @@ namespace ccf
       consensus->periodic_end();
     }
 
-    void node_msg(const std::vector<uint8_t>& data)
+    void recv_node_inbound(const uint8_t* payload_data, size_t payload_size)
     {
-      // Only process messages once part of network
-      if (
-        !sm.check(State::partOfNetwork) &&
-        !sm.check(State::partOfPublicNetwork) &&
-        !sm.check(State::readingPrivateLedger))
-      {
-        return;
-      }
-
-      const uint8_t* payload_data = data.data();
-      size_t payload_size = data.size();
-
       NodeMsgType msg_type =
         serialized::overlay<NodeMsgType>(payload_data, payload_size);
       NodeId from = serialized::read<NodeId::Value>(payload_data, payload_size);
 
-      switch (msg_type)
+      if (msg_type == ccf::NodeMsgType::forwarded_msg)
```

Comment:
Is it awkward to process forwarded messages but not channel messages? For example, if a forwarded message caused the channel to be re-established on the forwarder node, this node (the forwardee) wouldn't ever receive it?

Reply:
This is the same behaviour we had previously, just moved from the message handler in `enclave.h` into this function call.

I don't follow the example. A legitimate forwarded message will only ever arrive after channel establishment messages, which will only be accepted once this node is part of network. If we get a forwarded message before then, we will try to process it here but in practice fail to decrypt it later.

Reply:
I didn't realise this behaviour already existed - it's now a lot more obvious that forwarded messages are processed even though the node isn't in state `partOfNetwork/partOfPublicNetwork/readingPrivateLedger`. 

I was thinking about a node which transitioned from one of these 3 states to another state e.g. a new `retired` state (but such transitions don't actually exist, and these three states are final or transition to one another).